compiler-core/src/javascript.rs

```diff
@@ -172,22 +194,29 @@ impl<'a> Generator<'a> {
         // Put it all together
 
         if imports.is_empty() && statements.is_empty() {
-            Ok(docvec![type_reference, "export {}", line()])
+            Ok(docvec![
+                type_reference,
+                "export {}",
+                line(),
+                sourcemap_reference,
```

Comment:
The declaration of the SourceMap URL does not have to be at a specific location, it can be anywhere in the JS code. It is something that's separate from the type reference (which is just TS-specific in my understanding), so I think it's best to keep them as separate declaration blocks for symmetry. 

I take it from this comment that you'd rather have the sourcemap declaration at the top, so I moved it :)